pegasus/helpers/section_api_helpers.rb

```diff
@@ -526,38 +526,73 @@ def self.fields
 end
 
 class DashboardUserScript
+  # Assigns a script to all users enrolled in the section, creating a new user_scripts object if
+  # necessary. The method noops for those user_scripts that already exist with assigned_at set.
+  # WARNING: This method does not verify that the section and student_users exist (aren't deleted).
   def self.assign_script_to_section(script_id, section_id)
-    # create userscripts for users that don't have one yet
-    Dashboard.db[:user_scripts].
-      insert_ignore.
-      import(
-        [:user_id, :script_id],
-        Dashboard.db[:followers].
-          select(:student_user_id, script_id.to_s).
-          where(section_id: section_id, deleted_at: nil)
-      )
+    student_user_ids = Dashboard.db[:followers].
+      select(:student_user_id).
+      where(section_id: section_id, deleted_at: nil).
+      map {|f| f[:student_user_id]}
+    DashboardUserScript.assign_script_to_users(script_id, student_user_ids)
```

Comment:
As a meta question, is there a reason we have all this logic in pegasus, vs. adding something to dashboardapi like we've done for other places where we're accessing dashboard DB from pegasus?